### PR TITLE
add backup upload command

### PR DESCRIPTION
**Pull request by MaximilianMeister**
....and adapt base request class a bit

### CODE REVIEW

**Comment by tboerger** [comment]
Generally good, but you should check if the file really exists.

**Comment by MaximilianMeister** [comment]
...hmm any idea why travis fails? I have tried it even with a valid tarball but it never matches the expectation. even though there is no visible diff

```
#<Crowbar::Client::Request::Party (class)> received :post with unexpected arguments
         expected: ("/utils/backups/upload", {:headers=>{"Content-Type"=>"application/json", "Accept"=>"application/json"}, :query=>{:file=>#<File:/dev/null>}})
              got: ("/utils/backups/upload", {:headers=>{"Content-Type"=>"application/json", "Accept"=>"application/json"}, :query=>{:file=>#<File:/dev/null>}})
       Diff:
```

File: lib/crowbar/client/app/backup.rb [hunk 1]
```diff
@@ -97,11 +97,27 @@ def create(name)
           `download NAME CREATED_AT` will download a backup from the Administration Server.
```
Comment (tboerger) [inline lib/crowbar/client/app/backup.rb]: It still got CREATED_AT

File: lib/crowbar/client/app/backup.rb [hunk 2]
```diff
@@ -115,11 +131,10 @@ def download(name, created_at)
           `delete NAME CREATED_AT` will delete a backup from the Administration Server.
```
Comment (tboerger) [inline lib/crowbar/client/app/backup.rb]: CREATED_AT

File: spec/crowbar/client/request/backup/upload_spec.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../../../../spec_helper"
+
+describe "Crowbar::Client::Request::Backup::Upload" do
+  context "a request class upload" do
```
Comment (tboerger) [inline spec/crowbar/client/request/backup/upload_spec.rb]: I totally dislike that you have reimplemented the context. Please fix the predefined one within support folder.

**Comment by tboerger** [comment]
> ...hmm any idea why travis fails? I have tried it even with a valid tarball but it never matches the expectation. even though there is no visible diff

Use a real fixture file instead of `/dev/null`.

File: spec/crowbar/client/request/backup/upload_spec.rb (hunk 3 above)
Comment (MaximilianMeister) [inline spec/crowbar/client/request/backup/upload_spec.rb]: how would you best replace `body` with `query` [here](https://github.com/crowbar/crowbar-client/blob/master/spec/support/request_examples.rb#L48) ?

**Comment by tboerger** [comment]
I don't really like the download path thing but let's merge it.
:+1:

**Comment by rsalevsky** [comment]
+1

**Pull request merged**